radar_slam:

  # Topics
  pointCloudTopic: "/radar_enhanced_pcl"               # Point cloud data <!-- pc2_raw inlier_pc2 segmented -->
  imuTopic: "/vectornav/imu"                         # IMU data
  odomTopic: "/odom"                   # IMU pre-preintegration odometry, same frequency as IMU
  gpsTopic: "/ublox/fix"                   # GPS odometry topic from navsat, see module_navsat.launch file

  # Frames
  lidarFrame: ""
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"

  # GPS Settings
  # useGpsElevation: false                      # if GPS elevation is bad, set to "false"
  # gpsCovThreshold: 2.0                        # m^2, threshold for using GPS data
  # poseCovThreshold: 25.0                      # m^2, threshold for using GPS data
  
  # Export settings
  # savePCD: false                              # 
  # savePCDDirectory: "/Downloads/4DRadarSLAM/"        #

  # Sensor Settings
  # downsampleRate: 1                           # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1 
  # lidarMinRange: 1.0                          # default: 1.0, minimum lidar range to be used
  # lidarMaxRange: 1000.0                       # default: 1000.0, maximum lidar range to be used

  # IMU Settings
  imuAccNoise: 0.0022281160035059417 # 3.9939570888238808e-03
  imuGyrNoise: 0.00011667951042710442 # 1.5636343949698187e-03
  imuAccBiasN: 0.00011782392708033614 # 6.4356659353532566e-05
  imuGyrBiasN: 2.616129872371749e-06 # 3.5640318696367613e-05
  imuGravity: 9.80511
  imuRPYWeight: 0.01

  # Extrinsics (Lidar -> IMU)
  # extrinsicTrans: [0, 0, 0]
  extrinsicTrans: [-0.3176955976234, -0.13761019052125, 0.05898352725152]
  # extrinsicRot: [1, 0, 0,
  #                 0, -1, 0,
  #                 0, 0, -1]
  # extrinsicRPY: [1, 0, 0,
  #                 0, -1, 0,
  #                 0, 0, -1]
  extrinsicRot: [0.999735807578, -0.0215215701795, -0.0081643477385,
                 -0.02148120581797, -0.9997581134183, 0.00502853428037,
                 -0.00826995351904, -0.0048509797951, -0.99995400578406]
  extrinsicRPY: [0.999735807578, -0.0215215701795, -0.0081643477385,
                 -0.02148120581797, -0.9997581134183, 0.00502853428037,
                 -0.00826995351904, -0.0048509797951, -0.99995400578406]
  # extrinsicRot: [1, 0, 0,
  #                 0, 1, 0,
  #                 0, 0, 1]
  # extrinsicRPY: [1, 0, 0,
  #                 0, 1, 0,
  #                 0, 0, 1]


